x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_exploration/components/exploration_query_bar/exploration_query_bar.tsx

```diff
@@ -52,7 +52,7 @@ export const ExplorationQueryBar: FC<ExplorationQueryBarProps> = ({
     if (defaultQueryString !== undefined) {
       setSearchInput({ query: defaultQueryString, language: SEARCH_QUERY_LANGUAGE.KUERY });
     }
-  }, []);
+  }, [defaultQueryString !== undefined]);
```

Comment:
do you need to perform this check on every update?

Reply:
This check is required since `defaultQueryString` will be `undefined` on first render. I only need it to update once it has a value and then never again for the life of this component.